src/envelope.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+use crate::errors::{InternalPakeError, PakeError};
+use generic_array::{
+    typenum::{Unsigned, U32},
+    GenericArray,
+};
+use hkdf::Hkdf;
+use hmac::{Hmac, Mac, NewMac};
+use rand_core::{CryptoRng, RngCore};
+use sha2::{Digest, Sha256};
+
+// Constant string used as salt for HKDF computation
+const STR_ENVU: &[u8] = b"EnvU";
+
+/// The length of the "export key" output by the client registration
+/// and login finish steps
+pub(crate) type ExportKeySize = <Sha256 as Digest>::OutputSize;
+
+/// This struct is a straightforward instantiation of the trait separating the
+/// three components in Vecs
```

Comment:
I think we need a docstring that's far less "internal dev notes" and more specific in its reference to the spec. In particular, here, we are taking the "storage/communication efficient EnvU" option described in the spec in §4, and should say so IMHO. Moreover, there's no more trait :D